### DISCUSSION TITLE
brew tap and Docker :latest "releases" for non-@latest versions

### DISCUSSION THREAD

**Discussion by myitcv** [goreleaser/goreleaser#3114] [Q&A] [answered]
Hi all,

I've had a search through issues/discussions and haven't turned up anything on this topic. 

On the [CUE project](https://cuelang.org/) we use GoReleaser to build not on [binary release artefacts on GitHub](https://github.com/cue-lang/cue/blob/87408d7d0c61bf6ff2711fd202727639aac97290/.goreleaser.yml#L6), but also a [Homebrew Tap](https://github.com/cue-lang/cue/blob/87408d7d0c61bf6ff2711fd202727639aac97290/.goreleaser.yml#L49) and [Docker images](https://github.com/cue-lang/cue/blob/87408d7d0c61bf6ff2711fd202727639aac97290/.goreleaser.yml#L62). Notice the Docker manifests also [update the `:latest` tag](https://github.com/cue-lang/cue/blob/87408d7d0c61bf6ff2711fd202727639aac97290/.goreleaser.yml#L98).

However, we see a problem when creating pre-release builds (very common) or patch versions of earlier minor versions (less common but will start to happen). i.e. any version that does not correspond to the [`@latest`](https://go.dev/ref/mod#version-queries) at that point in time (according to the VCS for the module, not the proxy).

Our GoReleaser config fires on every release, but to my understanding we only want the Homebrew Tap part to fire when that version corresponds to the [`@latest` version query](https://go.dev/ref/mod#version-queries). Same applies for updating the `:latest` tag on Docker. 

I couldn't find anything in the documentation regarding this - have I missed something?

We plan to start maintaining our GoReleaser config as CUE (and export Yaml when we come to run GoReleaser) so we could work around this and maintain the conditional parts of this configuration in CUE, but it feels like this should be more of a first-class concept in GoReleaser.

Any thoughts greatly appreciated.

**Comment by caarlos0** [comment, marked as answer]
hey! yes, currently this is not possible... although I think it would be doable - at least for github, not sure about gitlab et al

thanks for raising this

**Reply by myitcv** [reply]
Thanks for your response as ever, @caarlos0!

We are experimenting with a CUE-based solution for now and will share news of that here (we had plans to move to a CUE source of truth in any case).
